internal/status/status.go

```diff
@@ -85,6 +80,17 @@ func getRootStatusJson(ctx *log.Context, statusType types.StatusType, c types.Cm
 	return b, nil
 }
 
+// getSingleStatusItem returns a single status item for the given status type, command, and message.
+// This is useful when only a single status item is needed for an immediate status report.
+func GetSingleStatusItem(ctx *log.Context, statusType types.StatusType, c types.Cmd, msg string) (types.StatusItem, error) {
+	ctx.Log("message", "creating json to report status")
+	statusReport := types.NewStatusReport(statusType, c.Name, msg)
+	if len(statusReport) != 1 {
+		return types.StatusItem{}, errors.New("expected a single status item")
+	}
+	return statusReport[0], nil
```

Comment:
Why do we keep the previous statuses?

Reply:
We are not keeping previous statuses here. I'm just returning the first elemento of the status report because I am reusing the `types.NewStatusReport` method. That method returns a list of statuses with only a single item.